test/web-auth/web-auth.test.js

```diff
@@ -410,6 +418,56 @@ describe('auth0.WebAuth', function() {
       ); // eslint-disable-line
     });
 
+    it('should fail with an invalid state (null transaction)', function(done) {
+      var webAuth = new WebAuth({
+        domain: 'mdocs.auth0.com',
+        redirectUri: 'http://example.com/callback',
+        clientID: '0HP71GSd6PuoRYJ3p',
+        responseType: 'token'
+      });
+
+      var data = webAuth.parseHash(
+        {
+          hash: '#state=123&access_token=VjubIMBmpgQ2W2&id_token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczovL21kb2NzLmF1dGgwLmNvbS8iLCJzdWIiOiJhdXRoMHw0QVpERjU2Nzg5IiwiYXVkIjoiMEhQNzFHU2Q2UHVvUllKM0RYS2RpWENVVWRHbUJidXAiLCJleHAiOjE0Nzg1NjIyNTMsImlhdCI6MTQ3ODUyNjI1M30.LELBxWWxcGdYTaE_gpSmlNSdcucqyrhuHQo-s7hTDBA&token_type=Bearer&refresh_token=kajshdgfkasdjhgfas'
```

Comment:
How can be so sure that you won't break this in the future? We need to explicitly tell in this test that the transaction is null so it's clearer to which value you're comparing 123 against.

Reply:
fixed